src/python/tests/gff3/test_simplifier.py

```diff
@@ -0,0 +1,365 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `ensembl.io.genomio.gff3.restructure` module."""
+
+from contextlib import nullcontext as does_not_raise
+from os import PathLike
+from pathlib import Path
+from typing import Callable, ContextManager, List, Optional
+
+from Bio.SeqFeature import SeqFeature
+import pytest
+from pytest import param, raises
+
+from ensembl.io.genomio.gff3.exceptions import GFFParserError
+from ensembl.io.genomio.gff3.simplifier import GFFSimplifier
+
+
+def check_one_feature(input_gff: PathLike, output_gff: PathLike, check_function: str) -> SeqFeature:
+    """Load 1 feature from a GFF, apply a function, then write it back to a GFF."""
+    simp = GFFSimplifier()
+    simp.records.from_gff(input_gff)
+    # Get the only feature
+    feat = simp.records[0].features[0]
+    # Apply the named function
+    check_method = getattr(simp, check_function)
+    new_feat = check_method(feat)
+
+    if new_feat is not None:
+        # Put it back
+        simp.records[0].features = [new_feat]
+        simp.records.to_gff(output_gff)
+    return new_feat
+
+
+@pytest.mark.parametrize(
+    "in_gff, expected_gff, expectation",
+    [
+        param("ok_gene.gff", "ok_gene.gff", does_not_raise(), id="ok gene"),
+        param("lone/transcript.gff", "lone/transcript_simped.gff", does_not_raise(), id="lone transcript"),
+        param("lone/trna.gff", "lone/trna_simped.gff", does_not_raise(), id="lone tRNA"),
+        param("lone/rrna.gff", "lone/rrna_simped.gff", does_not_raise(), id="lone rRNA"),
+    ],
+)
+def test_create_gene_for_lone_transcript(
+    data_dir: Path,
+    tmp_dir: Path,
```

Comment:
```suggestion
    tmp_path: Path,
    data_dir: Path,
```
I really need to find out the root of this `tmp_dir` fixture and remove it. Note I have also swapped the order because it is usually more convenient to put "external" fixtures first.

Apply the same in all the other tests below.